File: programming/c/tp_40/copy.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char* argv[]) {
    if (argc < 3) {
        printf("Il faut passer deux noms de fichier en argument");
        return -1;
    }
    FILE* src = fopen(argv[1], "r");
    FILE* dest = fopen(argv[2], "w");

    char next;
    while (fscanf(src, "%c", &next) != EOF) {
        if (next == 'a') next = 'e';
        if (next == 'e') next = 'a';
        fprintf(dest, "%c", next);
    }

    fclose(src);
    fclose(dest);
}

File: programming/ocaml/tp/tp_40/copy.ml
let () =
  if Array.length Sys.argv < 3 then print_string "Il faut passer deux noms de fichier"
  else
    let n1 = open_in Sys.argv.(1) in
    let n2 = open_out Sys.argv.(2) in

    try
      while true do
        let next = input_line n1 in
        Printf.fprintf n2 "%s\n" next
      done
    with
    | End_of_file -> print_string "Copie terminée."
